libass/ass_render.c

```diff
@@ -1607,7 +1607,9 @@ wrap_lines_smart(ASS_Renderer *render_priv, double max_text_width)
                         ((s3 - 1)->bbox.xMax + (s3 - 1)->pos.x) -
                         (w->bbox.xMin + w->pos.x));
 
-                    if (DIFF(l1_new, l2_new) < DIFF(l1, l2)) {
+                    if (DIFF(l1_new, l2_new) < DIFF(l1, l2) && w > text_info->glyphs) {
```

Comment:
Shouldn’t this rather remove the first line (and join it with the second line) if w is the first glyph? Roughly like this:

```c
if (DIFF(l1_new, l2_new) < DIFF(l1, l2)) {
    if (w == text_info->glyphs || w->linebreak)
        text_info->n_lines--;
    if (w != text_info->glyphs)
        w->linebreak = 1;
    s2->linebreak = 0;
    exit = 0;
}
```

Regarding the commit message: in addition to producing an unwanted empty line at the start, I think setting `text_info->glyphs->linebreak` also causes a [buffer overread](https://github.com/grigorig/libass/blob/b72b283b936a600c730e00875d7d067bded3fc26/libass/ass_render.c#L1473-L1477) in `trim_whitespace`—probably the cause of #229.